src/swap/utils/sendSingleTransactions.ts

```diff
@@ -0,0 +1,59 @@
+import type { TransactionReceipt } from 'viem';
+import { waitForTransactionReceipt } from 'wagmi/actions';
+import type { SendSingleTransactionsParams } from '../types';
+
+export async function sendSingleTransactions({
+  config,
+  sendTransactionAsync,
+  transactions,
+  updateLifecycleStatus,
+}: SendSingleTransactionsParams) {
+  let transactionReceipt: TransactionReceipt | undefined;
+
+  // Execute the non-batched transactions sequentially
+  for (let i = 0; i < transactions.length; i++) {
+    const tx = transactions[i];
+    updateLifecycleStatus({
+      statusName: 'transactionPending',
+    });
+    const txHash = await sendTransactionAsync(tx);
+
+    if (transactions.length === 3) {
+      // Permit2 has 3 transactions, 2nd to last is the `Permit2` approval
+      if (i === transactions.length - 3) {
+        updateLifecycleStatus({
+          statusName: 'transactionApproved',
+          statusData: {
+            transactionHash: txHash,
+            transactionType: 'Permit2',
+          },
+        });
+      }
+    }
+    // 2nd to last transaction is the `ERC20` approval
+    if (i === transactions.length - 2) {
+      updateLifecycleStatus({
+        statusName: 'transactionApproved',
+        statusData: {
+          transactionHash: txHash,
+          transactionType: 'ERC20',
+        },
+      });
```

Comment:
Going forward, we may want to persist these individual transaction approvals so lifecycleState ends up as [{transactionHash, transactionType}, ...]

Reply:
let's discuss with Leo offline and will add in a future PR - relatively trivial change

Reply:
Oh I see, `transactionHash` get overridden each time.

I important is for us to store those over time? Or is this more information we give to the devs and the devs can decide to store or not?

In the Transaction expeirence we use the word `transactionHashList: Address[];`, which i can see here not quite working because the different types of transaction.

@0xAlec @alessey so first question to agree on, are those transactionHash, important to store?

Reply:
besides just being transparent as to what is going on behind the scenes, i'm not really sure what you would do with them.